src/resolve/resolved-bus.c

```diff
@@ -1580,6 +1582,132 @@ static int bus_method_reset_server_features(sd_bus_message *message, void *userd
         return sd_bus_reply_method_return(message, NULL);
 }
 
+static int bus_method_register_service(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        _cleanup_(dnssd_service_freep) DnssdService *service = NULL;
+        _cleanup_free_ char *path = NULL;
+        _cleanup_free_ char *instance_name = NULL;
+        Manager *m = userdata;
+        DnssdService *s = NULL;
+        DnsTxtItem *last = NULL;
+        int r;
+
+        assert(message);
+        assert(m);
+
+        if (m->mdns_support != RESOLVE_SUPPORT_YES)
+                return sd_bus_error_set_errno(error, EPROTONOSUPPORT);
+
+        service = new0(DnssdService, 1);
+        if (!service)
+                return sd_bus_error_set_errno(error, ENOMEM);
+        service->manager = m;
+
+        r = sd_bus_message_read(message, "sssqqq", &service->name,
+                                &service->name_tpl,
+                                &service->type, &service->port,
+                                &service->priority, &service->weight);
+        if (r < 0)
+                return sd_bus_error_set_errno(error, -r);
+
+        s = hashmap_get(m->dnssd_services, service->name);
+        if (s)
+                return sd_bus_error_setf(error, BUS_ERROR_DNSSD_SERVICE_EXISTS, "DNS-SD service '%s' exists already", service->name);
+
+        r = dnssd_render_instance_name(service, &instance_name);
+        if (r < 0)
+                return sd_bus_error_set_errno(error, -r);
```

Comment:
the sd-bus library is fine with negative error, it will negate implicitly where necessary, that means no need to negative here explictly you can replace "-r" by just "r"

Reply:
(and everywhere else similar)